packages/react-auth/src/shared/session.ts

```diff
@@ -0,0 +1,140 @@
+import { APIResponse, AuthUser, CookieOptions, Token } from "@src/types";
+import { isLocalhost } from "./utils";
+
+type CookieObj = {
+  [key: string]: string;
+};
+
+export const SESSION_DATA_KEY = "pangea-authn";
+export const DEFAULT_COOKIE_OPTIONS: CookieOptions = {
+  cookieMaxAge: 60 * 60 * 24 * 2, // 48 Hours, in seconds
+  cookieName: SESSION_DATA_KEY,
+};
+
+const BASE_COOKIE_FLAGS = "; path=/";
+
+/*
+  Session storage functions
+*/
+
+export function getStorageAPI(isUsingCookies: boolean): Storage {
+  // Depending on if we're using cookies or not, we want to use a different storage Web API
+  // When we use a cookie, use session storage
+  // When we do not use a cookie, use local storage <--- default
+  return isUsingCookies ? window.sessionStorage : window.localStorage;
+}
+
+export const getToken = (cookieName: string, storageAPI?: Storage) => {
+  if (storageAPI === undefined) {
+    const cookies = getCookies();
+
+    return cookies[cookieName];
+  }
+
+  const data = getSessionData(storageAPI);
+  return data?.active_token?.token;
+};
+
+export const getSessionData = (storageAPI = localStorage) => {
+  const data = storageAPI.getItem(SESSION_DATA_KEY);
+  const session_json = data ? JSON.parse(data) : {};
+
+  return session_json;
+};
+
+const mapTokenData = (data: any) => {
+  const tokenData: Token = (({
```

Comment:
this seems a little strange, just trying to remove email and profile, with a shallow copy you could just delete them explicitly

Reply:
yea, good suggestion... at first i was only going to copy a couple values. removed the copy, deleting two keys